W-19626762 Agent Visualizer Overview (#2)
mulesoft/docs-agent-visualizer: modules/ROOT/pages/view-and-navigate-agent-network.adoc

+
[%header,cols="30a,70a"]
|===
| Field | Description
| Name | Name of the asset. 
| Description | Description of the asset.
|Provider | Provider of the asset.
| Platform | Platform of the asset, for example, Agentforce or MuleSoft.
| LLM | LLM provider (for example, `OpenAI`).
| Skills | Skills that are supported by the agent. Skills are concrete, callable operations within a capability, with defined input and output. For example,`lookupOrder`, `createCase`, `updateAddress`.
| Linked Agents | Agents that are linked to the broker. 
|MCP Servers and Tools | MCP servers and tools that are connected to the broker.
| Enabled Capabilities | Capabilities that are supported by the agent. Capabilities are high-level areas of what an agent can do (domains it supports, for example, customer support or knowledge search). 
| Instance Name | Instance name as it appears in API Manager.
| Environment Type | Environments the instance is deployed to, for example, `sandbox` or `production`.
| Based on Asset Version | Version of the instance. 
| Policies | Policies that are applied to the instance.
| Governance and Security | Shows whether the asset is protected by a Flex Gateway.

> | Performance and Health a| Shows the performance and health metrics for the instance including:

Reviewer: The metrics in Monitoring are different. Adding below.

* Successful Requests
+

> This metric is shown for brokers, agents, and MCP servers. This is the total number of requests received successfully in the last 24 hours. 

Reviewer: ```suggestion
* Total Requests
+
This metric is shown for brokers, agents, and MCP servers. This is the total number of requests received successfully in the last 24 hours. 

* Failed Requests
+
This metric is shown for brokers, agents, and MCP servers. This is the total number of unsuccessful requests in the last 24 hours. 
```

For Failed Requests, those are requests to the broker, agent, or MCP server. It's clear in the UI, but what I added is not as clear here. 

I used descriptions like this:
https://beta.docs.mulesoft.com/beta-agent-fabric/monitoring/anypoint-insights-agentic

Glad I rechecked Monitoring. Gonna fix my MCP server one, which refers/referred to agents. 😬

Reviewer: All these _default_ to 24 hours in Monitoring. They can change that once in Monitoring.

* Failed Request Rate
+
This metric is shown for brokers, agents, and MCP servers. This is the percentage of requests that failed in the last 24 hours. 

> * A2A Calls Sent
> +
> This metric is shown for brokers. This is the total number of messages sent to agents using the Agent2Agent (A2A) protocol in the last 24 hours. 
> * MCP Calls Sent
> +
> This metric is shown for MCP servers. This is the total number of messages sent to  MCP servers in the last 24 hours. 

Reviewer: These broker ones won't be implemented in Monitoring (yet)
```suggestion
```

> = View and Navigate Your Agent Network

Reviewer 2: ```suggestion
= Viewing and Navigating Your Agent Network
```

> Explore an interactive map of agents, brokers, and Model Context Protocol (MCP) servers. Pan and zoom, search and filter by environment or type, toggle layers, and select nodes or edges to view detailed metadata, metrics, and links for deeper investigation.

Reviewer 2: MCP again

> == Accessing Agent Visualizer

Reviewer 2: ```suggestion
== Access Agent Visualizer
```

> You need these Anypoint Platform permissions to access Agent Visualizer:

Reviewer 2: Suggest using a partial since this is identical with the overview.

* Anypoint Monitoring
* Monitoring: Monitoring Viewer
* Exchange Viewer
* API Manager View: View API Configuration, View Policies

. Log in to Anypoint Platform.
. On the Anypoint Platform home page, click *Open Agent Visualizer*.
+
If you don't see the agent network architecture, it's because there are no discoverable agentic assets in Anypoint Exchange. 

For more information about creating an agent network project and publishing assets to Exchange, see xref:anypoint-code-builder::index.adoc[].

> == Navigating the Agent Network

Reviewer 2: ```suggestion
== Navigate the Agent Network
```

> Agent Visualizer provides an interactive view of your agent network where you can:

Reviewer 2: For later - it might be nice to have a picture with the JTBD as callouts.

* Zoom in and out. 
* Click and drag on the canvas to pan the visualization.
* Search for the name of an asset on the canvas to highlight the nodes that contain your search terms.
* Turn on layers to see additional metadata or specific views.
* Filter the graph by design view (as defined in the broker YAML), environment, or agent type (broker, Agentforce agents, external agents, or MCP servers).
* Navigate through interactive elements using the Tab key.
* Navigate between connected and disconnected nodes using arrow keys when the canvas has focus.

> == Viewing the Agent Network

Reviewer 2: ```suggestion
== View the Agent Network
```

When you land on the Agent Visualizer page, you see the agent network map showing all discovered agentic assets nodes and their relationships (edges).

> If the page is empty, it means there are no discoverable agentic assets in Anypoint Exchange. For more information about creating an agent network project and publishing assets to Exchange, see xref:anypoint-code-builder::index.adoc[].

Reviewer 2: ```suggestion
If the page is empty, there are no discoverable agentic assets in Anypoint Exchange. For more information about creating an agent network project and publishing assets to Exchange, see xref:anypoint-code-builder::index.adoc[].
```

> === Viewing Data Flow Layers

Reviewer 2: ```suggestion
=== View Data Flow Layers
```

> Data flow (traffic) layers are used to group nodes in the agent network visualization to show production or non-production traffic. Only one type of traffic layer (production or non-production) is visible at one time.

Reviewer 2: ```suggestion
Data flow (traffic) layers are used to group nodes in the agent network visualization to show production or nonproduction traffic. Only one type of traffic layer (production or nonproduction) is visible at one time.
```

By default, the production data flow layer is shown. 

> Use the toggle to switch between the default view of production traffic and the view of non-production traffic (development/sandbox traffic).

Reviewer 2: ```suggestion
Use the toggle to switch between the default view of production traffic and the view of nonproduction traffic (development/sandbox traffic).
```

> image::agent-visualizer-layers-toggle.png[Data Flows toggle is turned to view the Production Environments layer]

Reviewer 2: ```suggestion
image::agent-visualizer-layers-toggle.png[Data Flows toggle selected to view the Production Environments layer]
```

Reviewer 2: or "turned on"?

Reviewer 2: Should "Production Environments" be lowercase here?

> When the non-production data flow layer is active, traffic between nodes in development or sandbox environments is shown as solid edges, and edges representing production traffic disappear. 

Reviewer 2: ```suggestion
When the nonproduction data flow layer is active, traffic between nodes in development or sandbox environments is shown as solid edges, and edges representing production traffic disappear. 
```

> == Filtering the Agent Network

Reviewer 2: ```suggestion
== Filter the Agent Network
```

Reviewer 2: Should this be Filter the Agent Network Dataset?

> You can filter the data set to focus your view.

Reviewer 2: ```suggestion
To filter the dataset to focus your view:
```

. Select the filter option.

> . Choose the filter criteria such as Agent type (broker, internal agents, external agents) or specific platforms (for example,Agentforce).

Reviewer 2: ```suggestion
. Select the filter criteria, such as agent type (broker, internal agents, external agents) or specific platforms (for example, Agentforce).
```

> . Choose an environment view to narrow the graph to nodes deployed in that environment.

Reviewer 2: ```suggestion
. Select an environment view to narrow the graph to nodes deployed in that environment.
```

[[viewing-node-details]]

> == Viewing Node Details

Reviewer 2: ```suggestion
== View Node Details
```

When you select a node, a detail card opens to provide comprehensive information for the asset.

> . Select any node in the agent network (or press Enter/Spacebar after focusing the node with arrow keys) to display its detail panel.

Reviewer 2: ```suggestion
. Select any node in the agent network (or press Enter or Spacebar after focusing the node with arrow keys) to show its detail panel.
```

. Review the node details.
+
[NOTE]
Broker metadata is defined in the agent network YAML. For more information about creating an agent network project, see xref:anypoint-code-builder::index.adoc[].
+
[%header,cols="30a,70a"]
|===
| Field | Description
| Name | Name of the asset. 
| Description | Description of the asset.
|Provider | Provider of the asset.
| Platform | Platform of the asset, for example, Agentforce or MuleSoft.
| LLM | LLM provider (for example, `OpenAI`).

> | Skills | List of skills that are supported by the agent. Skills are concrete, callable operations within a capability, with defined input and output. For example,`lookupOrder`, `createCase`, `updateAddress`.

Reviewer 2: ```suggestion
| Skills | Skills that are supported by the agent. Skills are concrete, callable operations within a capability, with defined input and output. For example,`lookupOrder`, `createCase`, `updateAddress`.
```

> | Linked Agents | List of agents that are linked to the broker. 

Reviewer 2: ```suggestion
| Linked Agents | Agents that are linked to the broker. 
```

> |MCP Servers and Tools | List of MCP servers and tools that are connected to the broker.

Reviewer 2: ```suggestion
|MCP Servers and Tools | MCP servers and tools that are connected to the broker.
```

> | Enabled Capabilities | Lists the capabilities that are supportted by the agent. Capabilities are high-level areas of what an agent can do (domains it supports, for example customer support, knowledge search). 

Reviewer 2: ```suggestion
| Enabled Capabilities | Capabilities that are supported by the agent. Capabilities are high-level areas of what an agent can do (domains it supports, for example, customer support or knowledge search). 
```

| Instance Name | Instance name as it appears in API Manager.

> | Environment Type | Lists the environments the instance is deployed to, for example, `sandbox` or `production`.

Reviewer 2: ```suggestion
| Environment Type | Environments the instance is deployed to, for example, `sandbox` or `production`.
```

> | Based on Asset Version | The version of the instance. 

Reviewer 2: ```suggestion
| Based on Asset Version | Version of the instance. 
```

> | Policies | Lists the policies that are applied to the instance.

Reviewer 2: ```suggestion
| Policies | Policies that are applied to the instance.
```

| Governance and Security | Shows whether the asset is protected by a Flex Gateway.
| Performance and Health a| Shows the performance and health metrics for the instance including:

* Successful Requests
+

> This metric is shown for brokers, agents, and MCP servers. This is the total number of requests received successfully in the last 24 hours. 

Reviewer 2: Can you reuse this content from troubleshoot?

If not, see comments there, starting on line 9.

* Failed Request Rate
+
This metric is shown for brokers, agents, and MCP servers. This is the percentage of requests that failed in the last 24 hours. 
* A2A Calls Sent
+
This metric is shown for brokers. This is the total number of messages sent to agents using the Agent2Agent (A2A) protocol in the last 24 hours. 
* MCP Calls Sent
+
This metric is shown for MCP servers. This is the total number of messages sent to  MCP servers in the last 24 hours. 

> | Monitoring | Provides links to logs and traces for quick access to troubleshooting tools. 

Reviewer 2: ```suggestion
| Monitoring | Links to logs and traces for quick access to troubleshooting tools. 
```

> | Tags| All the tags associated with the asset in Exchange.

Reviewer 2: ```suggestion
| Tags| All tags associated with the asset in Exchange.
```

|===

[[viewing-edge-details]]

> == Viewing Edge Details

Reviewer 2: ```suggestion
== View Edge Details
```

When you select an edge, the detail panel displays relationship information and metrics. Metrics are aggregated for all instances that exist between nodes. 

. Select an edge to open its detail panel.

> . Examine the metrics.

Reviewer 2: ```suggestion
. Examine the metrics:
```

= Viewing and Navigating Your Agent Network

Explore an interactive map of agents, brokers, and Model Context Protocol (MCP) servers. Pan and zoom, filter by environment, toggle layers, and select nodes or edges to view detailed metadata, metrics, and links for deeper investigation.

[[access-agent-visualizer]]
== Access Agent Visualizer

include::partial$partial-agent-visualizer.adoc[tag=agent-visualizer-permissions]

. Log in to Anypoint Platform.

> . On the Anypoint Platform home page, click *Open Agent Visualizer*.

Reviewer 2: For future (and future-proofing), just say "Click *Open Agent Visualizer*" (or just "Open Agent Visualizer").

+

> When you land on the Agent Visualizer page, you see the agent network map showing all discovered agentic assets nodes and their relationships (edges).

Reviewer 2: ```suggestion
On the Agent Visualizer page, you see the agent network map showing all discovered agentic assets, nodes and their relationships (edges).
```

[[navigate-agent-network]]
== Navigate the Agent Network

Agent Visualizer provides an interactive view of your agent network where you can zoom in and out, click and drag on the canvas to pan the visualization, and turn on environment layers (production or nonproduction) to see additional views.

[[view-data-flow-layers]]
== View Data Flow Layers

Data flow (traffic) layers are used to group nodes in the agent network visualization to show production or nonproduction traffic. Only one type of traffic layer (production or nonproduction) is visible at one time.

By default, the production data flow layer is shown. 

Use the toggle to switch between the default view of production traffic and the view of nonproduction traffic (development/sandbox traffic).

image::agent-visualizer-layers-toggle.png[Data Flows toggle is turned on to view the production environments layer]

When the nonproduction data flow layer is active, traffic between nodes in development or sandbox environments is shown as solid edges, and edges representing production traffic disappear. 

[[view-node-details]]
== View Node Details

> When you select a node, a detail card opens to provide comprehensive information for the asset.

Reviewer 2: details?

+
[%header,cols="30a,70a"]
|===
| Field | Description
| Name | Name of the asset. 
| Description | Description of the asset.
|Provider | Provider of the asset.
| Platform | Platform of the asset, for example, Agentforce or MuleSoft.
| LLM | LLM provider (for example, `OpenAI`).
| Skills | Skills that are supported by the agent. Skills are concrete, callable operations within a capability, with defined input and output. For example,`lookupOrder`, `createCase`, `updateAddress`.
| Linked Agents | Agents that are linked to the broker. 
|MCP Servers and Tools | MCP servers and tools that are connected to the broker.
| Enabled Capabilities | Capabilities that are supported by the agent. Capabilities are high-level areas of what an agent can do (domains it supports, for example, customer support or knowledge search). 
| Instance Name | Instance name as it appears in API Manager.
| Environment Type | Environments the instance is deployed to, for example, `sandbox` or `production`.
| Based on Asset Version | Version of the instance. 
| Policies | Policies that are applied to the instance.

> | Governance and Security | Shows whether the asset is protected by a Flex Gateway.

Reviewer 2: ```suggestion
| Governance and Security | Whether the asset is protected by a Flex Gateway.
```

> | Performance and Health a| Shows the performance and health metrics for the instance including:

Reviewer 2: ```suggestion
| Performance and Health a| Performance and health metrics for the instance including:
```

* Successful Requests
+
This metric is shown for brokers, agents, and MCP servers. This is the total number of requests received successfully in the last 24 hours. 
* Failed Request Rate
+
This metric is shown for brokers, agents, and MCP servers. This is the percentage of requests that failed in the last 24 hours. 
* A2A Calls Sent
+
This metric is shown for brokers. This is the total number of messages sent to agents using the Agent2Agent (A2A) protocol in the last 24 hours. 
* MCP Calls Sent
+

> This metric is shown for MCP servers. This is the total number of messages sent to  MCP servers in the last 24 hours. 

Reviewer 2: ```suggestion
This metric is shown for MCP servers. This is the total number of messages sent to MCP servers in the last 24 hours. 
```
